File: chatwoot/csharp/src/OSEG.ChatwootExamples/UpdateAgentInAccountExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using Com.Chatwoot.Api;
using Com.Chatwoot.Client;
using Com.Chatwoot.Model;

namespace OSEG.ChatwootExamples;

public class UpdateAgentInAccountExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey.Add("api_access_token", "USER_API_KEY");

        var updateAgentInAccountRequest = new UpdateAgentInAccountRequest(
            role: UpdateAgentInAccountRequest.RoleEnum.Agent
        );

        try
        {
            var response = new AgentsApi(config).UpdateAgentInAccount(
                accountId: 0,
                id: 0,
                data: updateAgentInAccountRequest
            );

            Console.WriteLine(response);
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling AgentsApi#UpdateAgentInAccount: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: chatwoot/typescript-axios/src/UpdateAgentInAccountExample.ts
import axios, { AxiosError } from "axios";
import * as api from "chatwoot_client"

const configuration = new api.Configuration({
  apiKey: "USER_API_KEY",
});

const updateAgentInAccountRequest: api.UpdateAgentInAccountRequest = {
  role: api.UpdateAgentInAccountRequestRoleEnum.Agent,
};

new api.AgentsApi(configuration).updateAgentInAccount(
  0, // accountId
  0, // id
  updateAgentInAccountRequest, // data
).then(response => {
  console.log(response.data);
}).catch((error: Error | AxiosError) => {
  console.log("Exception when calling AgentsApi#updateAgentInAccount:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
